python/ffcc/io.py

```diff
@@ -172,26 +180,39 @@ def _list_subfolders(path):
     return subfolders
 
   folders = [os.path.join(path, f) for f in _list_subfolders(path)]
-  data = []
+  # include the main directory
+  folders.append(path)
+  image_files = []
   for folder in folders:
-    try:
-      illuminant, weight, frames = read_from_directory(folder)
-      for i, f in enumerate(frames):
-        rgb, extended_feature = f
-        data.append({
-            'name': '{}_{}'.format(os.path.basename(folder), i),
-            'rgb': rgb,
-            'extended_feature': extended_feature,
-            'illuminant': illuminant,
-            'weight': weight,
-        })
-    except:  # pylint: disable=bare-except
-      tf.logging.error('Unable to load %s', folder)
-
-  return data
-
-
-def write_dataset_to_tfrecord(dataset, output_dir):
-  """Encodes dataset into TFRecords and write into the destination folder."""
-  for d in dataset:
-    encode_to_tfrecord(d, output_dir)
+    if folder:
+      image_files = [
+        os.path.join(folder, file) for file in os.listdir(folder) if
+        file.endswith('.png')]
+
+  # check if cvfolds.txt file exists in the root path
+  if tf.io.gfile.exists(os.path.join(path, 'cvfolds.txt')):
+    cvfolds = np.loadtxt(os.path.join(path, 'cvfolds.txt'), dtype=np.int)
+  else:
+    # create a 3-fold cross-validation partition and store it in the root path
+    number_of_files = len(image_files)
+    indices = np.linspace(0, number_of_files - 1, number_of_files).astype(int)
+    cvfolds = np.zeros(number_of_files).astype(int)
+    random.shuffle(indices)
+    for fold in range(3):
+      cvfolds[indices[np.floor((fold * number_of_files) / 3).astype(int):
+                      np.floor(((fold + 1) * number_of_files) / 3).astype(int)
+              ]] = int(fold + 1)
```

Comment:
- This line is difficult to parse. Can you break down the indices[...] into a variable to index cvfolds by?
- You can condense np.floor(x / 3).asint() by using integer division directly: x // 3.
- int(fold + 1) -> you don't need 'int'; fold is already an integer.